src/main/java/org/sonarlint/intellij/util/SonarLintAppUtils.java

```diff
@@ -128,16 +129,18 @@ private static String getPathRelativeToProjectBaseDir(Project project, VirtualFi
   }
 
   /**
-   * Path will always contain forward slashes.
+   *  Path will always contain forward slashes. Resolving the module path uses the official alternative to
+   *  {@link com.intellij.openapi.module.Module#getModuleFilePath} which is marked as internally!
    */
   @CheckForNull
   private static String getPathRelativeToModuleBaseDir(Module module, VirtualFile file) {
-    var moduleFilePath = module.getModuleFilePath();
-    if ("".equals(moduleFilePath)) {
-      // Non persistent module
-      return null;
-    }
-    var baseDir = Paths.get(moduleFilePath).getParent();
+    var moduleContentRoots = Arrays.stream(ModuleRootManager.getInstance(module).getContentRoots())
+            .filter(contentRoot -> contentRoot.getPath().trim().length() > 0)
+            .toArray(VirtualFile[]::new);
+
+    // If there are multiple content roots (this can be possible based on the SDK API documentation), just take the
+    // first one, assuming this one is the path of the ".iml" file.
+    var baseDir = Paths.get(moduleContentRoots[0].getPath()).getParent();
```

Comment:
I have a doubt about this:
* why stopping at the first content root ? The `file` parameter could be located in another root
* I don't think we need to still call `getParent()` now that we get the exact content root

This part is not covered with tests today, so we could take the opportunity to cover the code a bit better, e.g. with light/heavy tests to verify the different scenarios

Reply:
Very good point, I'll reopen the ticket. I was assuming a bit too much, that it would be handled by tests and actually didn't check if there even were corresponding tests.